### PR TITLE
Added actions for lint and tests

### PR DESCRIPTION
**Pull request by mhutchinson**
Change dependabot github action update to daily for consistency with other repos.

### CODE REVIEW

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+jobs:
+  golangci:
+    name: lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # v3.3.1
+        with:
+          go-version: 1.19
```
Comment (roger2hk) [inline .github/workflows/golangci-lint.yml:16]: The go-version here doesn't align with the one in go test.

Comment (mhutchinson) [reply .github/workflows/golangci-lint.yml]: Bumped the go version to 1.20 across the repo. PTAL.

Comment (mhutchinson) [reply .github/workflows/golangci-lint.yml]: Scratch that, I'll just make it 1.19 and 1..20 to be consistent. We'll bump the go version separately.

**Review by roger2hk — APPROVED** [review]

**Comment by mhutchinson** [comment]
@roger2hk PTAL - I downgraded back to 1.19 because it caused other problems and should be handled separately.

**Review by roger2hk — APPROVED** [review]

**Comment by roger2hk** [comment]
> @roger2hk PTAL - I downgraded back to 1.19 because it caused other problems and should be handled separately.

LGTM, thanks!

**Pull request merged**